scripts/pythondistdeps.py

```diff
@@ -267,11 +276,18 @@ def normalize_name(name):
             # See https://bugzilla.redhat.com/show_bug.cgi?id=1791530
             normalized_name = normalize_name(dist.project_name)
 
+            # If we're processing an extras subpackage, check that the extras exists
+            if extras_subpackage and extras_subpackage not in dist.extras:
+                print(f"\nError: The package name contains an extras name `{extras_subpackage}` that was not found in the metadata.\n"
+                      "Check if the extras were removed from the project. If so, consider removing the subpackage and obsoleting it from another.")
```

Comment:
This is extremely ugly when it happens:

```
Processing files: python3-setuptools_scm+hovno-4.1.2-1.fc33.noarch
error: Dependency tokens must begin with alpha-numeric, '_' or '/': `hovno` that was not found in the metadata.
error: Dependency tokens must begin with alpha-numeric, '_' or '/': `hovno` that was not found in the metadata.
Provides: Check Error: If The an and another. consider contains extras from if it name obsoleting package project. python-setuptools_scm+hovno = 4.1.2-1.fc33 python3-setuptools_scm+hovno = 4.1.2-1.fc33 python3.9-setuptools_scm+hovno = 4.1.2-1.fc33 removed removing so subpackage the were
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Requires: Check Error: If The an and another. consider contains extras from if it name obsoleting package project. removed removing so subpackage the were


RPM build errors:
    Dependency tokens must begin with alpha-numeric, '_' or '/': `hovno` that was not found in the metadata.
    Dependency tokens must begin with alpha-numeric, '_' or '/': `hovno` that was not found in the metadata.
```

I suppose we should put `*** PYTHON_EXTRAS_NOT_FOUND_ERROR ***` on the stdout. And the actuall error message on the stderr, possibly with loud padding (something like newline + `'=' * 35 + ' ERROR ' + '=' * 35` + newline + the actual error message + newline)

Reply:
I like it

Reply:
The reason for the error to be loud is because it is not the error the packagers will see as the error causing the failure. They will see:

```
RPM build errors:
    Dependency tokens must begin with alpha-numeric, '_' or '/': *** PYTHON_EXTRAS_NOT_FOUND_ERROR ***
```

So the stderr message must be attractive to see.

Reply:
(Also, I've verified that the message printed on stderr will be in the log.) I'm done for today...

Reply:
How about `*** PYTHON_EXTRAS_NOT_FOUND_ERROR__SEE_STDERR ***` ?

Reply:
No preference